### PR TITLE
feat: Add bank redirect mapping to adyen and stripe

### PR DESCRIPTION
**Pull request by dracarys18**
## Type of Change

- [x] New feature

## Description

Added bank redirect to adyen and stripe.

## Motivation and Context

This PR adds Bank redirect payment method to adyen and stripe

## How did you test it?

Manual

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed submitted code

### CODE REVIEW

**Review by Sangamesh26 — DISMISSED** [review]

**Review by bernard-eugine — APPROVED** [review]

File: crates/router/src/connector/adyen/transformers.rs [hunk 1]
```diff
@@ -311,6 +335,74 @@ pub enum PaymentType {
...
     Affirm,
     Afterpaytouch,
+    Eps,
+    Ideal,
+    Giropay,
+    #[serde(rename = "directEbanking")]
+    Sofort,
+}
+
+pub struct AdyenTestBankNames<'a>(&'a str);
```
Comment (SanchithHegde) [inline crates/router/src/connector/adyen/transformers.rs:345]: We could use a better name, maybe remove `Test`?

**Review by SanchithHegde — COMMENTED** [review]
Other than that, looks good to me!

File: crates/router/src/connector/adyen/transformers.rs [hunk 2]
```diff
@@ -213,6 +218,25 @@ pub enum AdyenPaymentMethod {
...
+    Ideal(BankRedirectionWithIssuer),
+    Giropay(BankRedirectionPMData),
+    Sofort(BankRedirectionPMData),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct BankRedirectionPMData {
+    #[serde(rename = "type")]
+    payment_type: PaymentType,
```
Comment (SamraatBansal) [inline crates/router/src/connector/adyen/transformers.rs:231]: Why use `rename_all` when you are doing field level conversion.

File: crates/router/src/connector/adyen/transformers.rs (hunk 1 above)
Comment (dracarys18) [reply crates/router/src/connector/adyen/transformers.rs:345]: Adyen has different banknames for test and production hence the `Test` in the name.

**Review by SanchithHegde — DISMISSED** [review]

**Review by Sangamesh26 — DISMISSED** [review]

File: crates/router/src/connector/adyen/transformers.rs [hunk 3]
```diff
@@ -213,6 +218,25 @@ pub enum AdyenPaymentMethod {
     AfterPay(AdyenPayLaterData),
     AdyenKlarna(AdyenPayLaterData),
     AdyenAffirm(AdyenPayLaterData),
+    Eps(BankRedirectionWithIssuer),
+    Ideal(BankRedirectionWithIssuer),
+    Giropay(BankRedirectionPMData),
+    Sofort(BankRedirectionPMData),
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
```
Comment (SamraatBansal) [inline crates/router/src/connector/adyen/transformers.rs:227]: Do we need both `serialize` and `deserialize` over these structs
Also see to all the structs that have been implemented, remove wherever redundant

**Review by SamraatBansal — CHANGES_REQUESTED** [review]

**Review by Sangamesh26 — APPROVED** [review]
lgtm

**Review by SamraatBansal — APPROVED** [review]
Looks Good to me, with a note for some enums to be shifted to separate config files so that the `transformers.rs` is not cluttered

**Review by SanchithHegde — APPROVED** [review]

**Pull request merged**